core/common/src/main/java/alluxio/util/URIUtils.java

```diff
@@ -313,8 +313,14 @@ private static int normalizedHash(int hash, String s) {
       if (ch == '%') {
         // hash next 2 characters
         index++;
+        if (index >= s.length()) {
```

Comment:
<...path>% and <...path> have the same hash? is it the expected behavior?

Reply:
I wondering if just the StandardURI class is calling the wrong hash function. Most of the values created and hashed in StandardURI use functions that say they decode the value already (except for authority which seems to be using a java generated toString method), so I guess maybe just a normal hash should be used instead?

Reply:
no they won't be the same.  the % will be accounted for before the if (ch=='%') line.

Reply:
the only reason of this distinguish % is to make sure Capital chars after % is interpreted as url encoded interpretation -> so that %3F is considered the same as %3f  the other chars will be hash-ed in the same way java hashcode() does. But whether we should not use AlluxiURI as URI syntax is discussible and needs cleanup properly. Will not touch upon this part in this fix.